clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -1049,6 +1055,8 @@ scheduler_tmpl_t<RecordType, ReaderType>::get_initial_timestamps()
                 RecordType record = **input.reader;
                 if (record_type_is_timestamp(record, input.next_timestamp))
                     break;
+                if (record_type_is_instr(record))
```

Comment:
So if we find an instr before a timestamp we return STATUS_ERROR_INVALID_PARAMETER, instead of just reading the next timestamp? Explain why in comment.

Reply:
We assume there are no instructions in our queue in skip_instructions().  Plus this means the trace is malformed in some way, so the invalid param fits.  Added comments.